js/lib/cachetasks.js

```diff
@@ -119,19 +123,55 @@
      * @return {Boolean} True if it has the request, false otherwise.
      */
     hasCacheEntry: async(function*(request, cacheName) {
-      var cache = yield memoizedCaches.open(cacheName);
+      var cache = yield MemoizedCaches.open(cacheName);
       var response = yield cache.match(request);
       return (response) ? true : false;
     }),
+
+    refreshCacheEntry(request, cacheName, force = "") {
+      return async.task(function*() {
+        var cache = yield MemoizedCaches.open(cacheName);
+        var response = yield cache.get(request);
+        // If it's not stale and we are not forced to refresh
+        if (response && !force && !ResponseUtils.isStale(response)) {
+          return response;
+        }
+        if (force === "force" || !response) {
```

Comment:
i'm not a huge fan of using string values in place of booleans/options since they're prone to silent failures due to typos etc. :dancer: